File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int N = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();
        for(int i = 0; i < N; i++) {
            String S = br.readLine();
            if(S.equals("top")) {
                sb.append(stack.isEmpty() ? -1 : stack.peek()).append('\n');
            } else if(S.equals("size")) {
                sb.append(stack.size()).append("\n");
            } else if(S.equals("empty")) {
                sb.append(stack.isEmpty() ? 1 : 0).append("\n");
            } else if(S.equals("pop")) {
                sb.append(stack.isEmpty() ? -1 : stack.pop()).append("\n");
            } else {
                stack.push(Integer.parseInt(S.split(" ")[1]));
            }
        }
        System.out.println(sb.toString().trim());
    }
}

File: 백준/Silver/10828. 스택/스택.js
class Stack {
  constructor() {
    this.arr = [];
  }

  push(num) {
    this.arr.push(num);
  }

  pop() {
    return this.arr.pop();
  }

  peek() {
    return this.arr[this.arr.length - 1];
  }

  isEmpty() {
    return this.arr.length === 0 ? 1 : 0;
  }

  size() {
    return this.arr.length;
  }
}

let [N, ...arr] = require('fs').readFileSync('/dev/stdin').toString().split('\n').map(line => line.trim());
N = Number(N);
let stack = new Stack();
const ans = [];
for(let i = 0; i < N; i++) {
  if(arr[i] === "top") {
    ans.push(stack.isEmpty() ? -1 : stack.peek());
  } else if(arr[i] === "size") {
    ans.push(stack.size());
  } else if(arr[i] === "empty") {
    ans.push(stack.isEmpty());
  } else if(arr[i] === "pop") {
    ans.push(stack.isEmpty() ? -1 : stack.pop());
  } else {
    stack.push(Number(arr[i].split(' ')[1]));
  }
}
console.log(ans.join('\n'));